cpp/src/barretenberg/honk/pcs/ipa/ipa.hpp

```diff
@@ -156,34 +164,41 @@ template <typename Params> class InnerProductArgument {
     /**
      * @brief Verify the correctness of a Proof
      *
-     * @param vk Verification_key containing srs and pippenger_runtime_state to be used for MSM
-     * @param proof The proof containg L_vec, R_vec and a_zero
-     * @param pub_input Data required to verify the proof
+     * @param vk Verification_key contains the srs and pippenger_runtime_state to be used for MSM
+     * @param claim OpeningClaim contains the commitment, challenge, and the evaluation
+     * @param proof Proof contains L_vec, R_vec, and a_zero
+     * @param transcript Transcript contains the round challenges and the aux challenge
      *
      * @return true/false depending on if the proof verifies
      */
-    static bool reduce_verify(std::shared_ptr<VK> vk, const Proof& proof, const PubInput& pub_input)
+    static bool reduce_verify(std::shared_ptr<VK> vk,
```

Comment:
To follow the idiom established in KZG, we would use a class `ipa::VerificationKey` to avoid passing the vk as an argument here.. I see you defined this and tested it but that you don't use it. Apologies if I forgot a discussion on this--did you explain why you chose this approach?

Reply:
This problem stems from the fact that `ipa::VerificationKey` class does not take ownership of `SRS` and `pippenger_runtime_state` (we have an issue for this here https://github.com/AztecProtocol/barretenberg/blob/24ab0dca5c5df4173963863220a6777d35efbbbf/cpp/src/barretenberg/honk/pcs/commitment_key.hpp#L27). These two parameters are needed in `ipa::reduce_verify` for MSM unlike other PCSs. These parameter are set by calling a constructor in `commitment_key.test.cpp` [here](https://github.com/AztecProtocol/barretenberg/blob/24ab0dca5c5df4173963863220a6777d35efbbbf/cpp/src/barretenberg/honk/pcs/commitment_key.test.hpp#L54). Under this settings, we access these parameters by passing the `vk` shared ptr as argument to `ipa::reduce_verify` in `ipa_test.cpp` [here](https://github.com/AztecProtocol/barretenberg/blob/24ab0dca5c5df4173963863220a6777d35efbbbf/cpp/src/barretenberg/honk/pcs/ipa/ipa.test.cpp#L58). 
This was a due point in PR #205, and we decided to go with this keeping this as an issue (#236). In the current PR, we somehow solved the issue with `reduce_prove` by using the transcript. But I could not see a path to resolve it for `reduce_verify`. Please let me know if this makes sense. I shall record this in the issue #236.